File: src/main/java/io/github/tanguygab/tabadditions/spigot/TABAdditionsSpigot.java
package io.github.tanguygab.tabadditions.spigot;

import io.github.tanguygab.tabadditions.shared.TABAdditions;
import org.bukkit.plugin.java.JavaPlugin;

public class TABAdditionsSpigot extends JavaPlugin {

    public TABAdditionsSpigot() {
        TABAdditions.addProperties();
    }

    @Override
    public void onEnable() {
        TABAdditions.setInstance(new TABAdditions(new SpigotPlatform(this), this,getDataFolder()));
        TABAdditions.getInstance().load();
    }

    @Override
    public void onDisable() {
        TABAdditions.getInstance().disable();
    }

}

File: src/main/kotlin/io/github/tanguygab/tabadditions/spigot/TABAdditionsSpigot.kt
package io.github.tanguygab.tabadditions.spigot

import io.github.tanguygab.tabadditions.shared.TABAdditions
import org.bukkit.plugin.java.JavaPlugin

class TABAdditionsSpigot : JavaPlugin() {
    init {
        TABAdditions.addProperties()
    }

    override fun onEnable() {
        TABAdditions.INSTANCE = TABAdditions(SpigotPlatform(this), this, dataFolder)
        TABAdditions.INSTANCE.load()
    }

    override fun onDisable() {
        TABAdditions.INSTANCE.disable()
    }
}
